File: implementation/scipamato/core/persistence-jooq/src/intTest/java/ch/difty/scipamato/core/persistence/paper/searchorder/SearchTermEvaluatorIntegrationTest.java
package ch.difty.scipamato.core.persistence.paper.searchorder;

import ch.difty.scipamato.core.entity.search.SearchTerm;

@SuppressWarnings("WeakerAccess")
public abstract class SearchTermEvaluatorIntegrationTest<T extends SearchTerm> extends SearchTermEvaluatorTest<T> {

    protected static final long   ID             = 1;
    protected static final long   SC_ID          = 10;
    protected static final String FN             = "fn";
    protected final        int    searchTermType = getSearchTermType();

    protected abstract int getSearchTermType();

    protected abstract T makeSearchTerm(String rawSearchTerm);

}

File: implementation/scipamato/core/persistence-jooq/src/intTest/kotlin/ch/difty/scipamato/core/persistence/paper/searchorder/SearchTermEvaluatorIntegrationTest.kt
package ch.difty.scipamato.core.persistence.paper.searchorder

import ch.difty.scipamato.core.entity.search.SearchTerm

abstract class SearchTermEvaluatorIntegrationTest<T : SearchTerm> {

    protected abstract val searchTermType: Int

    protected abstract fun makeSearchTerm(rawSearchTerm: String): T

    companion object {
        const val ID: Long = 1
        const val SC_ID: Long = 10
        const val FN = "fn"
    }

}
